sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -757,7 +758,10 @@ private[sql] class SQLConf extends Serializable with CatalystConf with Logging {
 
   def variableSubstituteDepth: Int = getConf(VARIABLE_SUBSTITUTE_DEPTH)
 
-  def warehousePath: String = new Path(getConf(WAREHOUSE_PATH)).toString
+  def warehousePath: String = {
+    val path = new Path(getConf(WAREHOUSE_PATH))
+    FileSystem.get(path.toUri, new Configuration()).makeQualified(path).toString
```

Comment:
I should say I'm assuming the input must be a URI, because we have to allow users to disambiguate by specifying a scheme. And if we do that, then Windows paths can't be allowed. Of course, Windows file URIs should work, like `file:///C:/...` So yes I am certainly assuming this must be a URI, not a path, because I think we have to? If so, sure, documenting it sounds good.